extensions/amp-story/1.0/amp-story-open-page-attachment.js

```diff
@@ -38,13 +40,49 @@ export const buildOpenAttachmentElement = (element) =>
     </a>`;
 
 /**
- * Renders the open attachment UI affordance.
+ * @param {!Element} element
+ * @return {!Element}
+ */
+export const buildOpenInlineAttachmentElement = (element) =>
+  htmlFor(element)`
+    <a class="
+        i-amphtml-story-page-open-attachment i-amphtml-story-system-reset"
+        role="button" aria-label="Swipe up">
+      <div class="i-amphtml-story-inline-page-attachment-chip">
+        <div class="i-amphtml-story-inline-page-attachment-img"></div>
+        <div class="i-amphtml-story-inline-page-attachment-arrow"></div>
+      </div>
+    </a>`;
+
+/**
+ * Determines which open attachment UI to render.
+ * @param {!Window} win
  * @param {!Element} pageEl
  * @param {!Element} attachmentEl
  * @return {!Element}
  */
-export const renderPageAttachmentUI = (pageEl, attachmentEl) => {
-  const openAttachmentEl = buildOpenAttachmentElement(pageEl);
+export const renderPageAttachmentUI = (win, pageEl, attachmentEl) => {
+  const openImgAttr = attachmentEl.getAttribute('cta-image');
+  const attachmentHref = attachmentEl.getAttribute('href');
+  if (
+    isInlinePageAttachmentV2ExperimentOn(win) &&
+    !attachmentHref &&
+    openImgAttr
+  ) {
+    return renderInlinePageAttachmentUI(win, pageEl, attachmentEl);
+  } else {
+    return renderDefaultPageAttachmentUI(pageEl, attachmentEl);
```

Comment:
If we're going to maintain two templates (one with images, one without), let's update the `renderInlinePageAttachmentUI` method name with something that reflects what it builds: the attachment UI that has images. Both are for the inline attachment so the names are confusing.

Reply:
We're duplicating a lot of code that we won't be able to remove when launching the experiment. Is it possible to have only one template for the new UI, so we can remove the old one when the experiment is launched?

Reply:
Updated it to `renderPageAttachmentUiWithImages`

Reply:
Unsure about that second part-- the way I see it is that we currently have:
1. old CTA UI (for outlinks & when no images are provided for inline)
2. inline CTA UI with images

And soon we'll have:
1. old CTA UI (when no images are provided for inline)
2. inline CTA UI with images
3. new outlink UI

I don't believe we'll ever want to remove the old UI completely, since it will still be used when no images are provided in-line, at least from what I understand from Hong's slides.

When the experiment is launched fully, all we'll be doing is changing the conditional logic in `renderPageAttachmentUI` to render one of these 3 UIs.

Reply:
Gabriel does make a good point. It could work for this to be one template.
The template could default as no-image UI and append image nodes if they're specified.

I think it would be more lean after launch but a little more code up-front since we would need to build the no-image UI into the experiment.

Reply:
I think Gabriel's suggestion is good since it reduces code and branching.

Here is an example commit of how this could be one template if you would like to reference it:
https://github.com/ampproject/amphtml/pull/33235/commits/642d29550c006a1d29223fa2d9e3bceec2eb9535

Also added some code comments since our function is doing several things.

Reply:
Philip and I investigated this and the way to collapse the templates when there's going to be 3 CTA UIs would be overly complicated and may cause regressions with the existing animations or other regressions. So I think we should keep the structure we have for now, because changing it would add complexity to readability without much benefit.